harness/determined/common/util.py

```diff
@@ -141,3 +144,25 @@ def get_config_path() -> pathlib.Path:
         config_path = pathlib.Path.home().joinpath(".config")
 
     return config_path.joinpath("determined")
+
+
+def get_max_retries_config() -> urllib3.util.retry.Retry:
+    # Allow overriding retry settings when necessary.
+    # `DET_RETRY_CONFIG` env variable can contain `urllib3` `Retry` parameters,
+    # encoded as JSON.
+    # For example:
+    #  - disable retries: {"total":0}
+    #  - shorten the wait times {"total":10,"backoff_factor":0.5,"method_whitelist":false}
+
+    config_data = os.environ.get("DET_RETRY_CONFIG")
+    if config_data is not None:
+        config = json.loads(config_data)
+        return urllib3.util.retry.Retry(**config)
+
+    # Defaults.
+    try:
+        kwargs = {"allowed_methods": False}
+        return urllib3.util.retry.Retry(total=20, backoff_factor=0.5, **kwargs)
+    except TypeError:  # Support urllib3 prior to 1.26
+        kwargs = {"method_whitelist": False}
+        return urllib3.util.retry.Retry(total=20, backoff_factor=0.5, **kwargs)
```

Comment:
rewritten from before to use kwargs, so as to avoid possibility of `unused #type: ignore comments`

Reply:
(this didn't pass linting in CI so I undid it)